registry-authorization-agent/src/main/java/com/hortonworks/registries/schemaregistry/authorization/AuthorizationAgent.java

```diff
@@ -0,0 +1,149 @@
+package com.hortonworks.registries.schemaregistry.authorization;
+
+import com.hortonworks.registries.schemaregistry.*;
+import com.hortonworks.registries.schemaregistry.errors.SchemaBranchNotFoundException;
+import com.hortonworks.registries.schemaregistry.errors.SchemaNotFoundException;
+import org.apache.hadoop.security.authorize.AuthorizationException;
+
+import javax.ws.rs.core.SecurityContext;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+public interface AuthorizationAgent {
+
```

Comment:
We need to keep the current implementation. It helps to leave the business logic (the data processing part) and the authorizer decoupled. Also, the current approach is simple and easy to understand in case if we need to add new APIs.

Reply:
Regarding Authorizer, now it contains just one generic method.